api/envoy/extensions/matching/common_inputs/network/v3/network_inputs.proto

```diff
@@ -95,3 +96,27 @@ message TransportProtocolInput {
 // [#extension: envoy.matching.inputs.application_protocol]
 message ApplicationProtocolInput {
 }
+
+// Input that matches the authentication attributes for a downstream.
+// [#extension: envoy.matching.inputs.authenticated]
+message AuthenticatedInput {
+  enum Field {
+    // It is not allowed to use and an exception will be throw if it is used. On of the following
+    // fields should be chosen.
+    FIELD_TYPE_UNSPECIFIED = 0;
+
+    // The URIs in the SAN field of the certificate. Certificate with multiple URIs will be
+    // concatenated into a ','-delimited list.
+    URI_SAN = 1;
+
+    // The DNS entries in the SAN field of the certificate. Certificate with multiple DNS
+    // entries will be concatenated into a ','-delimited list.
+    DNS_SAN = 2;
+
+    // The subject field of the certificate in RFC 2253 format.
+    SUBJECT = 3;
+  }
+
+  // The field in the peer certificate which should be matched on.
+  Field field = 1 [(validate.rules).enum = {defined_only: true}];
```

Comment:
I'd be a bit concerned that this sets us up for having a lot of these auth properties represented as an enum which wouldn't be easy to extend in the future, I think normally we prefer oneof of messages over enums

@markdroth do you have thoughts re this?

Reply:
Yeah, it seems like a oneof would be better than an enum here.  But I think this entire question goes away if we just have a separate matcher for each type.